R/lockfiles.R

```diff
@@ -96,7 +96,80 @@
 #' Note that the `Name` field may be empty. In that case, a project-local Python
 #' environment will be used instead (when not directly using a system copy of Python).
 #'
+#' @inheritParams snapshot
+#' @inheritParams renv-params
+#'
+#' @param lockfile An `renv` lockfile; typically created by either
+#'   `lockfile_create()` or `lockfile_read()`.
+#'
+#' @param file A file path, or \R connection.
+#'
 #' @family reproducibility
-#' @name lockfiles
-#' @rdname lockfiles
+#' @name lockfile
+#' @rdname lockfile
 NULL
+
+#' @param libpaths The library paths to be used when generating the lockfile.
+#' @rdname lockfile
+#' @export
+lockfile_create <- function(type = settings$snapshot.type(project = project),
+                            libpaths = .libPaths(),
+                            packages = NULL,
+                            exclude = NULL,
+                            ...,
+                            project = NULL)
+{
+  project <- renv_project_resolve(project)
+
+  renv_lockfile_create(
+    project = project,
+    libpaths = libpaths,
+    type = type,
+    packages = packages,
+    exclude = exclude
+  )
+}
+
+#' @rdname lockfile
+#' @export
+lockfile_read <- function(file = NULL, ..., project = NULL) {
+  project <- renv_project_resolve(project)
+  file <- file %||% renv_paths_lockfile(project = project)
+  renv_lockfile_read(file = file)
+}
+
+#' @rdname lockfile
+#' @export
+lockfile_write <- function(lockfile, file = NULL, ..., project = NULL) {
+  project <- renv_project_resolve(project)
+  file <- file %||% renv_paths_lockfile(project = project)
+  renv_lockfile_write(lockfile, file = file)
+}
+
+#' @param remotes A named \R list, mapping package names to the remote
+#'   specifications to be recorded in the lockfile.
+#'
+#' @param repos A named vector, mapping \R repository names to their URLs.
+#'
+#' @rdname lockfile
+#' @export
+lockfile_modify <- function(lockfile,
+                            ...,
```

Comment:
Not used?

Reply:
Not used internally, but I thought this might be useful for end users (e.g. a way to explicitly set the repositories in a lockfile)

Reply:
Oh I meant the `...`

Reply:
Oh sorry. That's there intentionally so we can:

1. Force some arguments to be named,
2. Add new arguments (in any order) in a backwards compatible way.

Reply:
(but maybe I should add `renv_dots_check()` to warn about unused arguments)

Reply:
Yeah, I think if you use `...` for this purpose, you need to error if there's anything in them.

Reply:
Ok; I made that change on main.